src/orion/testing/algo.py

```diff
@@ -416,84 +400,83 @@ def get_id(point, ignore_fidelity=False, exp_id=None):
         # Experiment id is ignored
         assert get_id([1, 1, 1], exp_id=1) == get_id([1, 1, 1], exp_id=2)
 
-    @phase
-    def test_seed_rng(self, mocker, num, attr):
+    @pytest.mark.parametrize("seed", [123])
+    def test_seed_rng(self, seed: int):
         """Test that the seeding gives reproducibile results."""
-        algo = self.create_algo()
+        algo = self.create_algo(seed=seed)
 
-        seed = numpy.random.randint(10000)
-        algo.seed_rng(seed)
-        spy = self.spy_phase(mocker, num, algo, attr)
-        trials = algo.suggest(1)
-        trials[0].id != algo.suggest(1)[0].id
+        trial_a = algo.suggest(1)[0]
+        trial_b = algo.suggest(1)[0]
+        assert trial_a != trial_b
 
-        new_algo = self.create_algo()
-        new_algo.seed_rng(seed)
-        self.force_observe(algo.n_observed, new_algo)
-        assert trials[0].id == new_algo.suggest(1)[0].id
-
-        self.assert_callbacks(spy, num + 1, new_algo)
+        new_algo = self.create_algo(seed=seed)
+        assert new_algo.n_observed == algo.n_observed
+        trial_c = new_algo.suggest(1)[0]
+        assert trial_c == trial_a
 
-    @phase
-    def test_seed_rng_init(self, mocker, num, attr):
+    def test_seed_rng_init(self):
         """Test that the seeding gives reproducibile results."""
         algo = self.create_algo(seed=1)
```

Comment:
I think this test will not work properly now since the new `create_algo` implementation is executing `algo.seed_rng`. Because of this we cannot test that the algo was seeded properly at instantiation.

Reply:
`create_algo` only calls `seed_rng` if a seed is passed as a parameter.

Reply:
Does that address your concern?

Reply:
See in this case it will clearly call seed_rng, wheras this test is for seeding durint init, and thus seed_rng should not be called. It's problematic.

Reply:
I don't think that the "call `seed_rng` internally if seed is passed to constructor" behaviour is something we want to enforce, since algos don't have to take a seed as a constructor parameter.

Reply:
I guess I could skip the test if the algo under test doesn't have a `seed` constructor parameter..

Reply:
Fixed in 1ce04a9a8b14b2193000c9222880632cacf4e7b2

Reply:
So far yes they need to: https://github.com/Epistimio/orion/blob/develop/src/orion/algo/gridsearch.py#L116. GridSearch is the only situation where we don't use the seed so far.